src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -21046,6 +21075,42 @@ static void M(C c, Base b)
                 );
         }
 
+        [Fact]
+        public void MaybeNull_Property_InDeconstructionAssignment_UnconstrainedGeneric()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class C<T>
+{
+    [MaybeNull] public T P { get; set; } = default!;
+
+    void M(C<T> c, T t)
+    {
+        (c.P, _) = (t, 1);
```

Comment:
This test demonstrates 1. the unconstrained MaybeNull warning and 2. that the flow state "loses" to [MaybeNull], right?

Reply:
Yes. This test also shows a third thing: 3. accessing `c.P` doesn't produce a warning when it is accessed as an L-value.

---
In reply to: [324901880](https://github.com/dotnet/roslyn/pull/38610#discussion_r324901880) [](ancestors = 324901880)